### ISSUE TITLE
What are fault events?

### ISSUE THREAD

**Issue by richardquint** [PiSupply/PiJuice#213]
I'm still coming to grip with my PiJuice and beginning Python.  While playing with the PiJuice Status functions I got a result of which I don't know the meaning.  GetStatus "is fault" is true, but on the Settings page Fault reports no fault. Is there any way to get more info on faults and what could they possibly be? I'm attaching screenshots of my script, its output and the Settings page.
![status-fault script](https://user-images.githubusercontent.com/39503532/45766697-0fb58e00-bbed-11e8-9904-0e2d9a118a9d.png)
![output of status-fault](https://user-images.githubusercontent.com/39503532/45766701-12b07e80-bbed-11e8-9d72-eee0f187c260.png)
![pijuice settings window](https://user-images.githubusercontent.com/39503532/45766705-147a4200-bbed-11e8-8966-e8d7cdd55b31.png)

**Comment by tvoverbeek** [comment]
What is reported as Fault on the settings page comes from the output of pj.status.GetFaultStatus(),
but only 'battery profile invalid' and charging temperature faults are shown on the HAT pane.
For example my current output of pj.status.GetFaultStatus() is `{'data': {'button_power_off': True}, 'error': 'NO_ERROR'}`.
Definitely needs more documentation about the internals.

**Comment by richardquint** [comment]
Okay, I get the same result for GetFaultStatus. See the screenshot. Where is that documented? Not on the https://github.com/PiSupply/PiJuice/tree/master/Software page, but there I do see that

"GetStatus() ... is_fault is True if there faults or fault events waiting to be read or False if there is no faults and no fault events." What does that mean in this case?  

Are there any other undocumented PiJuice functions?
![status-fault output](https://user-images.githubusercontent.com/39503532/45772384-59f23b80-bbfc-11e8-8bd5-472f9ed54c35.png)

**Comment by tvoverbeek** [comment]
> Are there any other undocumented PiJuice functions?

Yes, just look at the available functions in pijuice.py (/usr/lib/python2.7/dist-packages/pijuice.py or
/usr/lib/python3.x/dist-packages/pijuice.py).
You can check the Firmware sources for exact details.
The 'isFault' flag in GetStatus() is a summary from all eventfaults ORed together:
From the firmware source:
```
powerOffBtnEventFlag;
forcedPowerOffFlag;
forcedVSysOutputOffFlag;
watchdogExpiredFlag;
((currentBatProfile == NULL) ? 0x20 : 0);
CHRGER_TS_FAULT_STATUS();
```
Only the last two are reported in the GUI/CLI
